.github/actions/install-build-deps/action.yml

```diff
@@ -22,41 +22,107 @@ inputs:
+      env:
+        GCS_BUCKET: gs://perfetto-ci-artifacts
+        CACHE_PATH: cache/buildtools-${{ steps.cachekey.outputs.flags_key }}/buildtools.tar
+        FILES_TO_IGNORE_PATH: ${{ steps.files_to_ignore.outputs.file_path }}
+        LOCAL_BUILDTOOLS_HASH: ${{ hashFiles('buildtools/**', 'tools/install-build-deps') }}
+        CACHED_BUILDTOOLS_HASH_FILE_PATH: buildtools/buildtools_hash_revision_for_cache_on_ci.txt
+      run: |
+        # Restore content of 'buildtools/' folder from cache.
+        CACHED_TAR_PATH=$(mktemp /tmp/buildtools-cache-XXXXXX.tar)
+        cp_output=""
+        if cp_output=$(gcloud storage cp "$GCS_BUCKET/$CACHE_PATH" "$CACHED_TAR_PATH" 2>&1); then
+          echo "$cp_output"
+        
+          # '--exclude-from' is used to not overwrite files or folders that are stored in git.
+          # '--keep-old-files' flag causes 'tar' to fail instead of overwriting files.
+          tar --keep-old-files --preserve-permissions --extract --exclude-from="$FILES_TO_IGNORE_PATH" --file="$CACHED_TAR_PATH" --directory=buildtools
+        
+          # Content of 'buildtools/' rarely changes, so we don't want to re-upload unchanged cache on each build.
+          # It changes when the 'tools/install-build-deps' or the files added to git inside `buildtools/' changes,
+          # so we calculate and save the sha256 hash of that files.
+          if [ -f "$CACHED_BUILDTOOLS_HASH_FILE_PATH" ]; then
+            CACHED_BUILDTOOLS_HASH=$(cat "$CACHED_BUILDTOOLS_HASH_FILE_PATH")
+          else
+            CACHED_BUILDTOOLS_HASH="0"
+          fi
+        
+          if [[ "$CACHED_BUILDTOOLS_HASH" != "$LOCAL_BUILDTOOLS_HASH" ]]; then
+            # Rewrite the hash file, it is saved as part of 'buildtools/' cache.
+            echo "$LOCAL_BUILDTOOLS_HASH" > "$CACHED_BUILDTOOLS_HASH_FILE_PATH"
+            echo "update_cache=true" >> "$GITHUB_OUTPUT" 
+          fi
+        
+          rm "$CACHED_TAR_PATH"
+        
+          echo "Cache restored, OK"
+        elif [[ "$cp_output" =~ "The following URLs matched no objects or files" ]]; then
+          echo "Cache not found, OK"
+        else
+          echo "Can't download cache: $cp_output"
+        fi
+
+    - name: Run install-build-deps
       shell: bash
+      # 'tools/install-build-deps' script will check the hashes and download the correct deps if the restored cache is too old or too new.
+      run: tools/install-build-deps ${{ inputs.install-flags }}
 
-    - name: Save build-deps cache (only on main)
+    - name: Save buildtools cache (only on main)
       if: github.ref == 'refs/heads/main'
-      uses: actions/cache/save@v4
-      with:
-        path: buildtools
-        # TODO(lalitm): see the comment above about the cache key.
-        key: buildtools-${{ hashFiles('buildtools/BUILD.gn') }}-${{ steps.cachekey.outputs.suffix }}-${{ hashFiles('tools/install-build-deps') }}
+      shell: bash
+      env:
+        GCS_BUCKET: gs://perfetto-ci-artifacts
```

Comment:
can this be a global env one level up? feeld weird that we have to set it twice, error prone

Reply:
We can't set an action level variable unfortunately, the best we can do is: c9e9fbb9ea65219d69028367ed2781b07dff5e62
Seems to be good enough for me.